dev-packages/application-package/src/application-package.ts

```diff
@@ -120,6 +119,27 @@ export class ApplicationPackage {
         return this._extensionPackages;
     }
 
+    /**
+     * All entry points defined by the installed Theia extensions.
+     *
+     * Map of entry point name to script import path.
+     */
+    get extensionEntryPoints(): ReadonlyMap<string, string> {
+        if (!this._extensionEntryPoints) {
+            this._extensionEntryPoints = new Map();
+            for (const extensionPackage of this.extensionPackages) {
+                for (const theiaExtension of extensionPackage.theiaExtensions) {
+                    if (typeof theiaExtension.entryPoints === 'object') {
+                        for (const [key, value] of Object.entries(theiaExtension.entryPoints)) {
```

Comment:
This code here requires entrypoint id's to be globally unique. I don't think this is a necessary or desirable requirement.